File: src/main/java/com/zxk/xxxt/POJO/Student.java
package com.zxk.xxxt.POJO;

import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

/**
 * 学生用户表
 */
@Data
public class Student {
    /**
    * 主键ID
    */
    private Integer id;

    /**
    * 登录账号，唯一
    */
    private String username;

    /**
    * 加密密码
    */
    private String password;

    /**
    * 邮箱
    */
    private String email;

    /**
    * 手机号
    */
    private String phone;

    /**
    * 头像URL
    */
    private String avatar;

    /**
    * 昵称
    */
    private String nickname;

    /**
    * 学号
    */
    private String studentId;

    /**
    * 学院
    */
    private String college;

    /**
    * 专业
    */
    private String major;

    /**
    * 年级
    */
    private String grade;

    /**
     * 状态：0-禁用，1-正常
     */
    @TableField("status")
    private Byte status;  // 数据库字段仍然是Byte

    /**
     * 信誉分：范围0-200，默认100
     */
    private Integer creditScore;

    /**
     * 创建时间（Unix时间戳，单位：秒）
     */
    private Long createTime;

    /**
    * 更新时间（Unix时间戳，单位：秒）
    */
    private Long updateTime;

    /**
    * 删除时间（Unix时间戳，单位：秒）
    * 0 表示未删除，有数值表示已删除
    */
    private Long deleteTime;

}

File: frontend/XXXTfrontEnd/src/api/student.ts
import request from '@/utils/request'

// 学生相关接口类型定义
export interface LoginDTO {
  unOrEmail: string
  password: string
}

export interface RegisterDTO {
  username: string
  studentId: string
  email: string
  password: string
  code: string
}

export interface Student {
  id: number
  username: string
  email: string
  phone?: string
  avatar?: string
  nickname?: string
  studentId: string
  college?: string
  major?: string
  grade?: string
  status: number
  creditScore: number
  createTime: number
  updateTime: number
}

export interface LoginResponse {
  accessToken: string
  refreshToken: string
  user: Student
}

// 学生API
export const studentApi = {
  // 登录
  login(data: LoginDTO) {
    return request.post<LoginResponse>('/student/login', data)
  },

  // 注册
  register(data: RegisterDTO) {
    return request.post<string>('/student/register', data)
  },

  // 发送验证码
  sendCode(email: string) {
    return request.post<string>('/student/getcode', null, {
      params: { email }
    })
  },

  // 获取当前用户信息
  getUserInfo() {
    return request.get<Student>('/student/userinfo')
  },

  // 更新用户信息
  updateUser(data: Partial<Student>) {
    return request.put<string>('/student/update', data)
  },

  // 修改密码
  changePassword(oldPassword: string, newPassword: string) {
    return request.put<string>('/student/password/change', {
      oldPassword,
      newPassword
    })
  },

  // 重置密码
  resetPassword(email: string, newPassword: string) {
    return request.put<string>('/student/password/reset', {
      email,
      newPassword
    })
  },

  // 检查用户名是否存在
  checkUsername(username: string) {
    return request.post<{ exists: boolean }>('/student/check/username', { username })
  },

  // 检查邮箱是否存在
  checkEmail(email: string) {
    return request.post<{ exists: boolean }>('/student/check/email', { email })
  },

  // 检查学号是否存在
  checkStudentId(studentId: string) {
    return request.post<{ exists: boolean }>('/student/check/studentId', { studentId })
  }
}

